### DISCUSSION TITLE
Refresh JWT Token in connection init

### DISCUSSION THREAD

**Discussion by andreadragotta** [enisdenjo/graphql-ws#484] [Q&A] [answered]
Hi, 

i'm using graphql-ws with URQL in Vue3, my configuration send via connection-init the JWT token but this is a short live token (5 minutes) after that time i want to refresh it but i don't know how i can do it. 

Anyone have a solution? Thank you

**Comment by enisdenjo** [comment, marked as answer]
There is a ["ws server and client auth usage with token expiration, validation and refresh" recipe](https://the-guild.dev/graphql/ws/recipes#ws-server-and-client-auth-usage-with-token-expiration-validation-and-refresh) showcasing my recommended approach.

Additionally, if curious, you can read [my opinion on authentication in WebSockets](https://github.com/enisdenjo/graphql-ws/discussions/292#discussioncomment-1864428).
